superbench/analyzer/generate_statistic.py

```diff
@@ -0,0 +1,200 @@
+            logger.error('output excel failed: {}'.format(str(e)))
+
+    def run(self, raw_data_file, output_dir, diagnosis_rule_file=None, summary_rule_file=None, digit=2, plot=False):
+        """Run the statistics generation.
+
+        Args:
+            raw_data_file (str): raw data file path
+            output_dir (str): output directory
+            diagnosis_rule_file (str): diagnosis rule file path
+            summary_rule_file (str): summary rule file path
+            digit (int): digit to round
+            plot (bool): whether to plot the data
+        """
+        try:
+            # aggregate results from different devices
+            self._raw_data_df = self.get_aggregate_data(raw_data_file, summary_rule_file)
+            # re-organize metrics by benchmark names
+            self._benchmark_metrics_dict = self._get_metrics_by_benchmarks(list(self._raw_data_df.columns))
+            # read existing baseline
+            baseline = {}
+            # read diagnosis rules
+            aggregated_df = self._raw_data_df.copy()
+            rules = file_handler.read_rules(diagnosis_rule_file)
+            if not self._parse_rules_and_baseline(rules, baseline):
+                logger.error('parse rule failed')
+                return None
+            else:
+                for rule in self._sb_rules:
+                    single_metric_rule = self._sb_rules[rule]
+                    metrics = list(single_metric_rule['metrics'].keys())
+                    function_name = self._sb_rules[rule]['function']
+                    rule_op = RuleOp.get_rule_func(DiagnosisRuleType(function_name))
+                    outputs = Parallel(n_jobs=-1)(
+                        delayed(self.fix_threshold_outlier_detection)
+                        (aggregated_df[metric], single_metric_rule, metric, rule_op) for metric in metrics
+                    )
+                    for index, out in enumerate(outputs):
+                        if not out:
+                            logger.error('Analyzer: filter healthy nodes failed')
+                            return
+                        aggregated_df[metrics[index]] = out[1]
+                        if plot:
+                            plot_steps(
+                                out[1].tolist(),
+                                title=metrics[index],
+                                save_path=os.path.join(
+                                    output_dir, 'figures', metrics[index].replace('/', '_').replace(':', '_') + '.png'
+                                ),
+                                show=False
+                            )
+            stat_df = self.calculate_statistics(aggregated_df)
+            excel_file = os.path.join(output_dir, 'benchmark_stability_stat.xlsx')
+            self.output_excel(excel_file, stat_df, digit)
+
+        except Exception as e:
+            logger.error('Analyzer: generate statisitics failed, msg: {}'.format(str(e)))
+
+
+if __name__ == '__main__':
```

Comment:
Please remove the main as well. Shall we separate PR for these two functions? We can merge generate_baseline first.